scripts/wrapper/withdraw.js

```diff
@@ -40,34 +42,18 @@ module.exports = function (web3, artifacts) {
     }
   }
 
-  const getMaxWithdrawableAmount = async function (
-    argv,
-    bracketAddress,
-    tokenData,
-    exchange,
-    currentBatchId,
-    printOutput = false
-  ) {
-    const log = printOutput ? (...a) => console.log(...a) : () => {}
+  const determineAmountToWithdraw = async function (argv, bracketAddress, tokenData, exchange) {
```

Comment:
This method is a little bit confusing to me. It does three very different things depending on some fields on the very general argv argument.

It seems that this logic is very tightly coupled to the method that gets invoked with this amount in the end (`buildRequestWithdraw`, `buildWithdraw`, `buildWithdrawAndTransferFundsToMaster`, etc).

Maybe having three different method and calling them inside this other if else block or even inside the `build...` methods would make this coupling more explicit?

Reply:
That's a good idea, it looks like there wouldn't be too much code duplication by doing that. I'll open an issue for that.

Reply:
Isn't this something that could be tackled in this PR. It seems like a bunch of special casing is only introduced here which creates this weird coupling.

Reply:
Maybe I don't understand your proposal, but as I read it I'd have to refactor most of the file to do what you want. It's not a large file, but it would multiply by three at least the number of lines changed.

Reply:
I'll leave this up as #222 since it's a nice small task for a cleanup session.